tests/test_connections/test_ledger/test_ledger.py

```diff
@@ -102,5 +93,12 @@ async def test_not_hanging(self) -> None:
         # sleep for `non_blocking_time + tolerance`
         await asyncio.sleep(non_blocking_time + tolerance)
 
-        # the normal task should be finished now
-        assert receive_task.done(), "Normal task is blocked by blocking task!"
+        # the normal task should be finished
+        assert normal_task.done(), "Normal task should be done at this point."
+
+        # the blocking task should not be done
+        assert not blocking_task.done(), "Blocking task should be still running."
+
+        # the receive task should not be done
+        # because it is only awaiting the `blocking_task`
+        assert not receive_task.done(), "Receive task is blocked by blocking task!"
```

Comment:
this is where we showcase the bug, a task is done and is waiting because receive is "ignoring" it